### ISSUE TITLE
Reject fetch if any request URL is http?

### ISSUE THREAD

**Issue by jakearchibald** [WICG/background-fetch#103]
Should we perform this check early?

**Comment by beverloo** [comment]
Rather than checking the protocol, as we'd want to allow requests to localhost, we could check whether `request` represents a [potentially thrustworthy origin](https://w3c.github.io/webappsec-secure-contexts/#is-origin-trustworthy)?

**Comment by jakearchibald** [comment]
I don't think we should reject early in this case. These checks currently happen in https://fetch.spec.whatwg.org/#concept-main-fetch step 4. There are many other pre-network-request checks. It'd be weird to do one of them, so we'd have to find a way to do all of them, which would be a significant refactor of the fetch spec.

I'm also worried rejection vs event will expose something that isn't currently exposed, but I don't have anything particular in mind.

I think our line in the sand should be "reject if the request constructor throws", plus our additional checks like if it's "no-cors".
